src/app/view/perspectives/table/body/rows/cell-group/data-cell/table-data-cell.component.html

```diff
@@ -11,14 +11,30 @@
   <ng-container
     *ngIf="data.attribute | cellDataInputInfo:data.dataValue:allowedPermissions: (editing$ | async) as dataInputInfo">
 
+<<<<<<< Updated upstream
     <ng-container *ngIf="!dataInputInfo.showDataInput">
       <div *ngIf="dataInputInfo.hasValue" class="text-pre overflow-hidden h-100 mx-1 align-middle"
            [innerHTML]="data.dataValue | previewDataValue | safeHtml"
+=======
+    <ng-container *ngIf="!dataInputInfo.showDataInput && dataInputInfo.hasValue">
+      <div *ngIf="data.attribute?.constraint?.type !== constraintType.Text; else richText"
+           class="text-pre overflow-hidden h-100 mx-1 d-flex align-items-center"
+>>>>>>> Stashed changes
            [title]="data.dataValue | formatDataValue | stripHtml:false"
            [ngClass]="data.attribute?.constraint | constraintClass"
            [class.invalid-value]="!(data.dataValue | isDataValueValid)"
            readonly data-test="table-data-input">
+        {{data.dataValue | previewDataValue}}
       </div>
+      <ng-template #richText>
+        <div [innerHTML]="data.dataValue | previewDataValue | safeHtml"
```

Comment:
What if user puts XSS text in Text constraint?

Reply:
The text input component encodes HTML characters in their & variants and it is displayed properly. I have tested this.